src/huggingface_hub/inference/_text_generation.py

```diff
@@ -0,0 +1,327 @@
+# coding=utf-8
+# Copyright 2023-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Original implementation taken from the `text-generation` Python client (see https://pypi.org/project/text-generation/
+# and https://github.com/huggingface/text-generation-inference/tree/main/clients/python)
+#
+# Changes compared to original implementation:
+# - use pydantic.dataclasses instead of BaseModel
+# - default to Python's dataclasses if Pydantic is not installed (same implementation but no validation)
+# - added default values for all parameters (not needed in BaseModel but dataclasses yes)
+# - integrated in `huggingface_hub.InferenceClient``
+# - added `stream: bool` and `details: bool` in the `text_generation` method instead of having different methods for each use case
+# - NO asyncio support yet => TODO soon
+
+from dataclasses import field
+from enum import Enum
+from typing import List, NoReturn, Optional
+
+from requests import HTTPError
+
+from ..utils import is_pydantic_available
+
+
+if is_pydantic_available():
+    from pydantic import validator
+    from pydantic.dataclasses import dataclass
+else:
+    # No validation if Pydantic is not installed
+    from dataclasses import dataclass  # type: ignore
+
+    def validator(x):  # type: ignore
+        return lambda y: y
+
+
+@dataclass
+class TextGenerationParameters:
+    # Activate logits sampling
+    do_sample: bool = False
+    # Maximum number of generated tokens
+    max_new_tokens: int = 20
+    # The parameter for repetition penalty. 1.0 means no penalty.
+    # See [this paper](https://arxiv.org/pdf/1909.05858.pdf) for more details.
+    repetition_penalty: Optional[float] = None
+    # Whether to prepend the prompt to the generated text
+    return_full_text: bool = False
+    # Stop generating tokens if a member of `stop_sequences` is generated
+    stop: List[str] = field(default_factory=lambda: [])
```

Comment:
I noticed we are not validating the stop sequences. I believe the max is 4

Reply:
Thanks for the heads up. @OlivierDehaene could you confirm this 4-items limit? I found the same in the [openapi.json](https://github.com/huggingface/text-generation-inference/blob/53aa9194c8c070afd19fa4660305dab2b280adf3/docs/openapi.json#L507) specs but prefer to cross-check with you

Reply:
`max_best_of` and `max_stop_sequences` are both parameters that can be modified in TGI. The defaults are 2 and 4 but they can also be turned off.

Reply:
Ok so let's not validate them client-side. Thanks for confirming